openedx/features/enterprise_support/utils.py

```diff
@@ -415,17 +415,16 @@ def is_enterprise_learner(user):
     Check if the given user belongs to an enterprise. Cache the value if an enterprise learner is found.
 
     Arguments:
-        user (User): Django User object or Django User object id.
+        user (User): Django User object.
 
     Returns:
         (bool): True if given user is an enterprise learner.
     """
-    user_id = user.id if isinstance(user, get_user_model()) else user
```

Comment:
FWIW, I suspect this was the problem, giving us the SimpleLazyObject that blows up a few lines later when it tries to filter `EnterpriseCustomerUser` by it. But given it only hit one of our envs, maybe this error only occurs when enterprise is not enabled for the environment?